create/application.go

```diff
@@ -102,10 +103,18 @@ func (app *applicationCmd) Run(ctx context.Context, client *api.Client) error {
 		if err := auth.Valid(); err != nil {
 			return fmt.Errorf("the credentials are given but they are empty: %w", err)
 		}
-		// for git auth we create a separate secret and then reference it in the app.
+
 		secret := auth.Secret(newApp)
-		if err := client.Create(ctx, secret); err != nil {
-			return fmt.Errorf("unable to create git auth secret: %w", err)
+		if err := client.Get(ctx, types.NamespacedName{Name: secret.Name, Namespace: secret.Namespace}, secret); err != nil {
+			// for git auth we create a separate secret and then reference it in the app.
+			if err := client.Create(ctx, secret); err != nil {
```

Comment:
I guess this should only happen when `if !errors.IsNotFound(err) {`?

Reply:
Or rather it might be easier to do `client.Create` and if that fails with `errors.IsAlreadyExists` we do the update.

Reply:
changed it to `errors.IsAlreadyExists` variant